### PR TITLE
Web update userinfo react and express vars

### PR DESCRIPTION
**Pull request by p-netm**
Web update userinfo react and express vars

### CODE REVIEW

**Review by dubdabasoduba — APPROVED** [review]

File: charts/opensrp-web/values.yaml [hunk 1]
```diff
@@ -89,7 +89,7 @@ containerEnvironmentVariables:
   EXPRESS_OPENSRP_CLIENT_SECRET: "{{ .Values.sharedVars.opensrpClientSecret }}"
   EXPRESS_OPENSRP_LOGOUT_URL: "https://{{ .Values.sharedVars.opensrpServer  }}/opensrp/logout.do"
   EXPRESS_OPENSRP_OAUTH_STATE: "{{ .Values.express.opensrpOauthState }}"
-  EXPRESS_OPENSRP_USER_URL: "https://{{ .Values.sharedVars.opensrpServer  }}/opensrp/user-details/"
+  EXPRESS_OPENSRP_USER_URL: "https://{{ .Values.sharedVars.keycloakUrl }}/auth/realms/{{ .Values.sharedVars.keycloakRealm }}/protocol/openid-connect/userinfo"
```
Comment (dubdabasoduba) [inline charts/opensrp-web/values.yaml:92]: @p-netm Does this work the same for OpenSRP Web and FHIR web?

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Review by machariamuguku — APPROVED** [review]

File: charts/opensrp-web/values.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ containerEnvironmentVariables:
   EXPRESS_OPENSRP_CLIENT_SECRET: "{{ .Values.sharedVars.opensrpClientSecret }}"
   EXPRESS_OPENSRP_LOGOUT_URL: "https://{{ .Values.sharedVars.opensrpServer  }}/opensrp/logout.do"
```
Comment (machariamuguku) [inline charts/opensrp-web/values.yaml:90]: do we need to also change this?

Comment (p-netm) [reply charts/opensrp-web/values.yaml:90]: That is ok, as it is, some deployments will still use this value as it is

File: charts/opensrp-web/values.yaml [hunk 3]
```diff
@@ -196,13 +199,14 @@ express:
...
       - "'self'"
-      - "https://github.com/OpenSRP/"
-      - "https://*.githubusercontent.com/OpenSRP/"
+      - "https://github.com/opensrp/"
+      - "https://*.githubusercontent.com/opensrp"
     connect-src:
       - "'self'"
       - "{{ if .Values.sentry.domain }}{{ .Values.sentry.domain }}{{ end }}"
+      - "{{ if .Values.sharedVars.fhirServer }}{{ .Values.sharedVars.fhirServer }}{{ end }}"
+      - "{{ if .Values.sharedVars.opensrpServer }}{{ .Values.sharedVars.opensrpServer  }}{{ end }}"
```
Comment (machariamuguku) [inline charts/opensrp-web/values.yaml:208]: `"{{ if .Values.sharedVars.opensrpServer }}{{ .Values.sharedVars.opensrpServer }}{{ else if .Values.sharedVars.fhirServer }}{{ .Values.sharedVars.fhirServer }}{{ end }}"`

Having one condition which is always falsey will result to an empty entry ""

Comment (machariamuguku) [reply charts/opensrp-web/values.yaml:208]: I actually don't think we need the `{{ end }}` block also but you can test it out

File: charts/opensrp-web/values.yaml [hunk 4]
```diff
@@ -145,10 +146,10 @@ reactEnvironmentVariables:
...
-  REACT_APP_OPENSRP_USER_URL: "https://{{ .Values.sharedVars.opensrpServer }}/opensrp/user-details/"
+  REACT_APP_OPENSRP_ROLES: '{"USERS": "ROLE_EDIT_KEYCLOAK_USERS,ROLE_VIEW_KEYCLOAK_USERS","PLANS": "ROLE_VIEW_KEYCLOAK_USERS","LOCATIONS": "ROLE_VIEW_KEYCLOAK_USERS","CARD_SUPPORT": "ROLE_VIEW_KEYCLOAK_USERS","INVENTORY": "ROLE_VIEW_KEYCLOAK_USERS","TEAMS": "ROLE_VIEW_KEYCLOAK_USERS","PRODUCT_CATALOGUE": "ROLE_VIEW_KEYCLOAK_USERS","FORM_CONFIGURATION": "ROLE_VIEW_KEYCLOAK_USERS","CARE_TEAM": "ROLE_VIEW_KEYCLOAK_USERS","SERVER_SETTINGS": "ROLE_VIEW_KEYCLOAK_USERS", "QUEST": "ROLE_VIEW_KEYCLOAK_USERS", "COMMODITY": "ROLE_VIEW_KEYCLOAK_USERS"}'
+  REACT_APP_OPENSRP_USER_URL: "{{ .Values.sharedVars.userDetailsUrl }}"
```
Comment (machariamuguku) [inline charts/opensrp-web/values.yaml:152]: some roles missing. Should we add all roles? eg HEALTHCARE_SERVICE

File: charts/opensrp-web/values.yaml [hunk 5]
```diff
@@ -89,7 +89,7 @@ containerEnvironmentVariables:
   EXPRESS_OPENSRP_CLIENT_SECRET: "{{ .Values.sharedVars.opensrpClientSecret }}"
   EXPRESS_OPENSRP_LOGOUT_URL: "https://{{ .Values.sharedVars.opensrpServer  }}/opensrp/logout.do"
   EXPRESS_OPENSRP_OAUTH_STATE: "{{ .Values.express.opensrpOauthState }}"
-  EXPRESS_OPENSRP_USER_URL: "https://{{ .Values.sharedVars.opensrpServer  }}/opensrp/user-details/"
+  EXPRESS_OPENSRP_USER_URL: "{{ .Values.sharedVars.userDetailsUrl }}"
```
Comment (machariamuguku) [inline charts/opensrp-web/values.yaml:92]: For some reason this does not resolve correctly. This is what is generated in the server: `EXPRESS_OPENSRP_USER_URL=https://{{ .Values.sharedVars.keycloakUrl }}/auth/realms/{{ .Values.sharedVars.keycloakRealm }}/protocol/openid-connect/userinfo`

File: charts/opensrp-web/values.yaml (hunk 4 above)
Comment (machariamuguku) [inline charts/opensrp-web/values.yaml:152]: For some reason this does not resolve correctly. This is what is generated in the server:

`REACT_APP_OPENSRP_USER_URL:"https://<no value>/auth/realms/<no value>/protocol/openid-connect/userinfo"`

**Review by machariamuguku — CHANGES_REQUESTED** [review]
I do not hate the idea of in lining the `.Values.sharedVars.userDetailsUrl` values

File: charts/opensrp-web/values.yaml (hunk 4 above)
Comment (p-netm) [reply charts/opensrp-web/values.yaml:152]: @machariamuguku Not necessarily, I do not think specifying for each module is how alot of instances will be setup, will more likely override it to just provide roles for the modules that a specific instance requires.

File: charts/opensrp-web/values.yaml (hunk 5 above)
Comment (p-netm) [reply charts/opensrp-web/values.yaml:92]: let me check

**Review by machariamuguku — APPROVED** [review]

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]
@p-netm Does this affect both OpenSRP web and FHIR Web? I think for OpenSRP web it should remain the same way. It should only change for FHIR web. 

At least the User details endpoint part.

**Comment by machariamuguku** [comment]
> @p-netm Does this affect both OpenSRP web and FHIR Web? I think for OpenSRP web it should remain the same way. It should only change for FHIR web.
> 
> At least the User details endpoint part.

@dubdabasoduba I think it's easier for deployments if both can get them from the same place. Helps make it dynamic and thus less configs for users. Also i think code has been changed for this flow

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**